backfill/2024-04-26-active_users_aggregates/fenix_query.sql

```diff
@@ -7,7 +7,7 @@ WITH baseline AS (
     OR attribution_source IS NOT NULL AS attributed,
     city,
     country,
-    um.distribution_id AS distribution_id,
+    att.distribution_id AS distribution_id,
```

Comment:
@bochocki `distribution_id` is null in unified_metrics. We can currently retrieve it from the clients table which returns the first non-null reported value for each client_id, let me know what you think of this source.

Reply:
For recent reporting, we've used `mozdata.fenix.metrics`, like below. Do you think there are any downsides to using that table, or additional benefits to using the clients table? We also recently noticed that both the clients table and the metrics table have incomplete `distribution_ids` ([Slack thread here](https://mozilla.slack.com/archives/C06LZ00TARL/p1716301230731579?thread_ts=1715967161.491459&cid=C06LZ00TARL)). So maybe we should use multiple sources?

```
SELECT
  client_info.client_id,
  "Fenix" as normalized_app_name,
  ARRAY_AGG(metrics.string.metrics_distribution_id)[SAFE_OFFSET(0)] AS distribution_id
FROM
  `mozdata.fenix.metrics`
```

Reply:
`[firefox_android_clients](https://github.com/mozilla/bigquery-etl/blob/4ebf436f057ffb5d4a9d73c4e56a217c19469b6a/sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v1/query.sql#L415-L425)` gets the first not-null `distribution_id` reported when looking at 3 pings: baseline, metrics & first session therefore it'd be expected to be the most complete one.

Reply:
Thanks for clarifying that, Lucia; I need a day to look into this.

Joy and I were discussing the best way to include the `distribution_id` in the client-level Fenix tables, since the baseline ping alone seems to have gaps in `distribution_id`. We are leaning towards only using the baseline ping for those tables, since it simplifies the pipeline and there's no "perfect" solution to the fact that different ping sources have different data.

I'm going to look into the impact on forecasts of using just the baseline ping vs `firefox_android_clients` for `distribution_id`. If the impact is minimal, it might be simplest to just use the baseline ping.

Sorry for the longer delay here.

Reply:
Thank you! Happy that we go for the best approach here.

Reply:
Changed to metrics ping as it retrieves more data (see sprint meeting notes).